tests/testthat/test-prepare_pctr_company.R

```diff
@@ -81,3 +81,42 @@ test_that("handles tiltIndicator output", {
     )
   )
 })
+
+test_that("yields a single distinct value of `*matching_certainty_company_average` per company", {
+  product <- tibble(
+    companies_id = c("id1", "id1"),
+    grouped_by = c("a", "a"),
+    risk_category = c("a", "a"),
+    clustered = c("building construction", "building construction"),
+    activity_uuid_product_uuid = c(
+      "ebd4dddf-9f74-5fd1-89ce-197b60cb8d06_006863b7-d736-4eb6-bbf8-648d292184ad",
+      "ebd4dddf-9f74-5fd1-89ce-197b60cb8d06_006863b7-d736-4eb6-bbf8-648d292184ad"
+    ),
+    co2_footprint = c("a", "a"),
+    tilt_sector = c("a", "a"),
+    tilt_subsector = c("a", "a"),
+    isic_4digit = c("a", "a")
+  )
+
+  company <- tibble(
+    companies_id = c("id1", "id1"),
+    grouped_by = c("a", "a"),
+    risk_category = c("high", "medium"),
+    value = c(1.0, 1.0)
+  )
+
+  result <- prepare_pctr_company(
+    company,
+    product,
+    ep_companies,
+    ecoinvent_activities,
+    matches_mapper,
```

Comment:
Here is a revised version of this test. See the comments:

```r
# This makes it clearer which are crucial inputs and makes them easier to reuse
id <- "id1"
clustered <- "building construction"
uuid <- "ebd4dddf-9f74-5fd1-89ce-197b60cb8d06_006863b7-d736-4eb6-bbf8-648d292184ad"

# `product` doesn't seem to need 2 duplicated rows.
product <- tibble(
  companies_id = id,
  grouped_by = "a",
  risk_category = "a",
  clustered = clustered,
  activity_uuid_product_uuid = uuid,
  co2_footprint = "a",
  tilt_sector = "a",
  tilt_subsector = "a",
  isic_4digit = "a",
)

# We can create a 2-row tibble more simply because it recycles the longest vector
company <- tibble(
  companies_id = id,
  grouped_by = "a",
  risk_category = c("high", "medium"),
  value = 1.0
)

result <- prepare_pctr_company(
  company,
  product,
  ep_companies,
  ecoinvent_activities,
  # matches_mapper is very large and makes tests extreemely slow. We can
  # use just the row we need to make it substantially faster.
  matches_mapper |> filter(activity_uuid_product_uuid == uuid),
  isic_tilt_mapper
)

# This tests has only one company so it makes no sense to `group_by(companies_id))`. If we
# are happy with one company, then we can simplify the expectation like this:
expect_equal(unique(result$matching_certainty_company_average), "high")

# It would be nice to test more than one company, but then the toy data we use here
# has be be extended
```